internal/notif/pushover/client.go

```diff
@@ -60,16 +72,77 @@ func (c *Client) Send(entry model.NotifEntry) error {
 		return err
 	}
 
-	_, err = pushover.New(token).SendMessage(&pushover.Message{
-		Title:     string(title),
-		Message:   string(body),
-		Priority:  c.cfg.Priority,
-		Sound:     c.cfg.Sound,
-		URL:       c.meta.URL,
-		URLTitle:  c.meta.Name,
-		Timestamp: time.Now().Unix(),
-		HTML:      true,
-	}, pushover.NewRecipient(recipient))
-
-	return err
+	cancelCtx, cancel := context.WithCancelCause(context.Background())
+	timeoutCtx, _ := context.WithTimeoutCause(cancelCtx, *c.cfg.Timeout, errors.WithStack(context.DeadlineExceeded)) //nolint:govet // no need to manually cancel this context as we already rely on parent
+	defer func() { cancel(errors.WithStack(context.Canceled)) }()
+
+	form := url.Values{}
+	form.Add("token", token)
+	form.Add("user", recipient)
+	form.Add("title", string(title))
+	form.Add("message", string(body))
+	form.Add("priority", strconv.Itoa(c.cfg.Priority))
+	if c.cfg.Sound != "" {
+		form.Add("sound", c.cfg.Sound)
+	}
+	if c.meta.URL != "" {
+		form.Add("url", c.meta.URL)
+	}
+	if c.meta.Name != "" {
+		form.Add("url_title", c.meta.Name)
+	}
+	form.Add("timestamp", strconv.FormatInt(time.Now().Unix(), 10))
+	form.Add("html", "1")
+
+	hc := http.Client{}
+	req, err := http.NewRequestWithContext(timeoutCtx, "POST", pushoverAPIURL, strings.NewReader(form.Encode()))
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Set("User-Agent", c.meta.UserAgent)
+
+	resp, err := hc.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.Header != nil {
+		var appLimit, appRemaining int
+		var appReset time.Time
+		if limit := resp.Header.Get("X-Limit-App-Limit"); limit != "" {
+			if i, err := strconv.Atoi(limit); err == nil {
+				appLimit = i
+			}
+		}
+		if remaining := resp.Header.Get("X-Limit-App-Remaining"); remaining != "" {
+			if i, err := strconv.Atoi(remaining); err == nil {
+				appRemaining = i
+			}
+		}
+		if reset := resp.Header.Get("X-Limit-App-Reset"); reset != "" {
+			if i, err := strconv.Atoi(reset); err == nil {
+				appReset = time.Unix(int64(i), 0)
+			}
+		}
+		log.Debug().Msgf("Pushover app limit: %d, remaining: %d, reset: %s", appLimit, appRemaining, appReset)
+	}
+
+	var respBody struct {
+		Status  int      `json:"status"`
+		Request string   `json:"request"`
+		Errors  []string `json:"errors"`
+		User    string   `json:"user"`
+		Token   string   `json:"token"`
+	}
+
+	if err = json.NewDecoder(resp.Body).Decode(&respBody); err != nil {
```

Comment:
Suggest logging the full unparsed response body (e.g. as a string) when debug logging turned on.

Reply:
And actually... On a second look, I'm not sure the existing log statement will work as intended. It's trying to serialize the struct, which won't be populated since JSON deserialization failed. I think you need to read the body as a string and log that instead.